Source/JavaScriptCore/API/tests/testapi.cpp

```diff
@@ -742,6 +757,258 @@ void TestAPI::testJSObjectSetOnGlobalObjectSubclassDefinition()
+    }
+
+    {
+        uint64_t uint64Min = std::numeric_limits<uint64_t>::min();
+        JSStringRef uint64MinStr = JSStringCreateWithUTF8CString("0x0");
+        const char* description = "checking JSBigIntCreateWithUInt64 with int64 min";
+        checkJSAndAPIMatch(
+            [&] {
+                return callFunction(createBigIntFunction, JSValueMakeString(context, uint64MinStr));
+            },
+            [&](JSValueRef* exception) {
+                JSValueRef bigint = JSBigIntCreateWithUInt64(context, uint64Min, exception);
+                checkIsBigIntType(bigint);
+                uint64_t result = JSBigIntToUInt64(context, bigint, exception);
+                check(result == uint64Min, description, ", result equals to uint64Min"_s);
+                return bigint;
+            },
+            description);
+
+        JSStringRelease(uint64MinStr);
+    }
+
+    {
+        const char* description = "checking JSBigIntToString with BigInt(0x2a) and toString(37)";
+        JSValueRef exceptionRef = nullptr;
+        JSValueRef* exception = &exceptionRef;
+        JSValueRef number = JSValueMakeNumber(context, 42.0);
+        uint64_t result = JSBigIntToUInt64(context, number, exception);
+        check(!result, description, " has correct result");
+        checkThrownException(exception, "ReferenceError: not a bigint JSValue"_s, description);
+    }
+
+    {
+        JSStringRef str = JSStringCreateWithUTF8CString("0x2a");
+        const char* description = "checking JSValueMakeBigIntFromString with 0x2a";
+        checkJSAndAPIMatch(
+            [&] {
+                return callFunction(createBigIntFunction, JSValueMakeString(context, str));
+            },
+            [&](JSValueRef* exception) {
+                JSValueRef bigint = JSBigIntCreateWithString(context, str, exception);
+                checkIsBigIntType(bigint);
+                JSStringRef string = JSBigIntToString(context, bigint, 16, exception);
+                check(string->string() == "2a"_s, "string should be \"2a\" (JSBigIntCreateWithString)");
+                JSStringRelease(string);
+                return bigint;
+            },
+            description);
+
+        JSStringRelease(str);
+    }
+
+    {
+        JSStringRef str = JSStringCreateWithUTF8CString("0h2a");
+        const char* description = "checking JSValueMakeBigIntFromString with 0h2a";
+        checkJSAndAPIMatch(
+            [&] {
+                return callFunction(createBigIntFunction, JSValueMakeString(context, str));
+            },
+            [&](JSValueRef* exception) {
+                JSBigIntCreateWithString(context, str, exception);
+                checkThrownException(exception, "SyntaxError: Failed to parse String to BigInt"_s, description);
+                return nullptr;
+            },
+            description);
+
+        JSStringRelease(str);
+    }
+
+    {
+        const char* description = "checking JSBigIntToString with BigInt(0x2a) and toString(16)";
```

Comment:
Nit: This descriptions could probably be updated.

Reply:
This description should be correct. But I did update other descriptions for testing `ReferenceError: not a bigint JSValue`.